client/client.go

```diff
@@ -251,17 +287,63 @@ func (tbc *tsoBatchController) pushRequest(tsoReq *tsoRequest) {
 	tbc.collectedRequestCount++
 }
 
-// adjustBestBatchSize stabilizes the latency with the AIAD algorithm.
-func (tbc *tsoBatchController) adjustBestBatchSize() {
-	tsoBestBatchSize.Observe(float64(tbc.bestBatchSize))
-	length := tbc.collectedRequestCount
-	if length < tbc.bestBatchSize && tbc.bestBatchSize > 1 {
-		// Waits too long to collect requests, reduce the target batch size.
-		tbc.bestBatchSize--
-	} else if length > tbc.bestBatchSize+4 /* Hard-coded number, in order to make `tbc.bestBatchSize` stable */ &&
-		tbc.bestBatchSize < tbc.maxBatchSize {
-		tbc.bestBatchSize++
+func (tbc *tsoBatchController) adjustBestBatchSize(maxBatchWaitInterval time.Duration) {
+	if maxBatchWaitInterval == 0 {
+		// maxBatchWaitInterval is 0, indicating that the strategy is closed.
+		tbc.statStartTime = typeutil.ZeroTime
+		return
+	}
+
+	if tbc.statStartTime == typeutil.ZeroTime {
+		// The strategy has just started, reset all statistics.
+		tbc.statStartTime = time.Now()
+		tbc.countTSOQuery = 0
+		tbc.countTSORequest = 0
+		tbc.countBatchWaitDuration = 0
+		tbc.countNotExtraTSOQuery = 0
+		tbc.bestBatchSize = 0
+		return
+	}
+
+	totalDuration := time.Since(tbc.statStartTime)
+	if totalDuration < tsoStatisticsInterval/2 {
+		// Ensure that the duration is long enough.
+		return
+	}
+
+	if tbc.countTSORequest < minCountTSORequest {
+		// If there are too few requests, turn off the batch strategy.
+		tbc.bestBatchSize = 0
+	} else if float64(tbc.countTSORequest)*float64(maxBatchWaitInterval) <= float64(tbc.countBatchWaitDuration)/(1+tsoStatisticsAmplification) {
```

Comment:
need more comments

Reply:
Yes, the most part of this PR LGTM but I think we still need some explanation on these calculations otherwise it's hard to maintain by other people.